src/GraphQL.Tests/Bugs/Bug1767InvalidByte.cs

```diff
@@ -0,0 +1,55 @@
+using GraphQL.SystemTextJson;
+using GraphQL.Types;
+using System;
+using System.Linq;
+using System.Numerics;
+using Xunit;
+
+namespace GraphQL.Tests.Bugs
+{
+    // https://github.com/graphql-dotnet/graphql-dotnet/pulls/1766
+    public class Bug1767InvalidByte : QueryTestBase<Bug1767Schema>
+    {
+        private void AssertQueryWithError(string query, string result, string message, int line, int column, object[] path, Exception exception = null, string code = null, string inputs = null)
+        {
+            var error = exception == null ? new ExecutionError(message) : new ExecutionError(message, exception);
+            if (line != 0) error.AddLocation(line, column);
+            error.Path = path;
+            if (code != null)
+                error.Code = code;
+            var expected = CreateQueryResult(result, new ExecutionErrors { error });
+            AssertQueryIgnoreErrors(query, expected, inputs?.ToInputs(), renderErrors: true, expectedErrorCount: 1);
+        }
+
+        [Fact]
+        public void Input_Byte_Valid_Variable() => AssertQuerySuccess("query($arg: Byte!) { input(arg: $arg) }", @"{ ""input"": ""2"" }", "{\"arg\":2}".ToInputs());
+
+        [Fact]
+        public void Input_Byte_Valid_Argument() => AssertQuerySuccess("query { input(arg: 2) }", @"{ ""input"": ""2"" }", "{\"arg\":2}".ToInputs());
+
+        [Fact]
+        public void Input_Byte_Invalid_Variable() => AssertQueryWithError("query($arg: Byte!) { input(arg: $arg) }", null, "Variable '$arg' is invalid. Unable to convert '300' to 'Byte'", 1, 7, null, null, "INVALID_VALUE", "{\"arg\":300}");
+
+        [Fact]
+        public void Input_Byte_Invalid_Argument() => AssertQueryWithError("query { input(arg: 300) }", null, "Argument \"arg\" has invalid value 300.\nExpected type \"Byte\", found 300.", 1, 15, null, code: "5.3.3.1");
```

Comment:
I think we should unify '' and "" inside messages.

Reply:
I agree, but there are many uses of single quotes and double quotes throughout the project.  It would need to be a separate PR.

Reply:
Yes.

Reply:
> Is it really necessary? Do we need `copyData` parameter?

No it is not strictly necessary.  I can execute `Data.Clear();` inside the `InvalidValueException` constructor instead.

> And why is it equal to `false` here?

This error is a syntax error, only occurring due to an invalid request.  Any information within the `Data` property could potentially be sensitive data and should not be returned to the client.  I'm not sure that the additional codes (e.g. `OVERFLOW`) should be returned either, but if you feel this is useful, that's fine with me.  Note that there are no other places where an ExecutionError is generated from infrastructure with an inner exception (besides unhandled exceptions), so this is the only spot where this constructor would be is used.

See also: https://github.com/graphql-dotnet/graphql-dotnet/issues/1641

I personally cannot see how at any time one would want the Data property to be copied.  So if it was me, I'd remove the `SetData` line of code from the `ExecutionError` constructor.  But at least this can be cleared within the `UnhandledExceptionDelegate` for unhandled exceptions.  (This particular error is a handled exception, not an unhandled exception.)

Let me know what you prefer - add the constructor, or use `Data.Clear();`.